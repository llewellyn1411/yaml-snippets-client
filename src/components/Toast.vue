<template>
    <div class="toast-wrapper" :class="{ active: isActive }">
        <div
            class="toast"
            :class="{ 'toast-success': type === 'success', 'toast-error': type === 'error', 'toast-warn': type === 'warning' }"
        >
            <button class="btn btn-clear float-right" @click="hide"></button>
            <h5 class="title">{{ title }}</h5>
            <p v-if="message" class="message">{{ message }}</p>
        </div>
    </div>
</template>

<script>
export default {
    name: 'Toast',
    props: {
        visible: {
            type: Boolean,
            default: false
        },
        title: {
            type: String,
            default: ''
        },
        message: {
            type: String,
            default: ''
        },
        type: {
            type: String,
            default: 'success'
        },
        onClose: {
            type: Function
        }
    },
    data() {
        return {
            isActive: false
        };
    },
    watch: {
        visible(value) {
            this.isActive = value;
        }
    },
    created() {
        this.isActive = this.visible;
    },
    methods: {
        hide() {
            this.isActive = false;
            this.onClose();
        }
    }
};
</script>
<style lang="scss" scoped>
.toast-wrapper {
    position: fixed;
    width: 100%;
    bottom: -200px;

    display: flex;
    justify-content: center;
    opacity: 0;

    transition: opacity 0.3s, bottom 0.3s;

    &.active {
        bottom: 50px;
        opacity: 1;
    }

    .toast {
        width: 640px;
    }
}
</style>
