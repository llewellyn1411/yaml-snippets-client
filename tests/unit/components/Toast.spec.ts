import { shallowMount } from '@vue/test-utils';
import Toast from '../../../src/components/Toast.vue';

describe( 'Toast.vue', () => {
    const closeFunc = jest.fn();
    const wrapper = shallowMount( Toast, {
        propsData: {
            visible: true,
            title: 'This is a toast',
            message: 'Do as it says',
            type: 'success',
            onClose: closeFunc
        }
    } );

    test( 'should show a toast', () => {
        expect( wrapper.find( '.toast-wrapper' ).exists() ).toBe( true );
        expect( wrapper.find( '.toast.toast-success' ).exists() ).toBe( true );
        expect( wrapper.find( '.title' ).text() ).toEqual( 'This is a toast' );
        expect( wrapper.find( '.message' ).text() ).toEqual( 'Do as it says' );
        expect( wrapper.html() ).toMatchSnapshot();
    } );

    test( 'should close a toast', () => {
        wrapper.find( 'button' ).trigger( 'click' );
        expect( closeFunc ).toHaveBeenCalled();
        expect( wrapper.find( 'toast-wrapper' ).exists() ).toBe( false );
    } );
} );
