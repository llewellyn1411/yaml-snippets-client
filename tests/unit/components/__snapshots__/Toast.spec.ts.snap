// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast.vue should show a toast 1`] = `
<div class="toast-wrapper active">
  <div class="toast toast-success"><button class="btn btn-clear float-right"></button>
    <h5 class="title">This is a toast</h5>
    <p class="message">Do as it says</p>
  </div>
</div>
`;
